apps/server/src/modules/files/job/sync-filerecords.console.ts

```diff
@@ -0,0 +1,62 @@
+import { EntityManager, ObjectId } from '@mikro-orm/mongodb';
+import { File, FileRecord, FileRecordParentType } from '@shared/domain';
+import { FileFilerecord } from '@shared/domain/entity/file_filerecord.entity';
+import { TaskRepo } from '@shared/repo';
+import { Logger } from '@src/core/logger';
+import { Command, Console } from 'nestjs-console';
+
+@Console({ command: 'sync-filerecords' })
+export class SyncFilerecordsConsole {
+	constructor(private taskRepo: TaskRepo, private em: EntityManager, private logger: Logger) {}
+
+	@Command({ command: 'tasks [batchSize]' })
+	async syncFilerecordsForTasks(batchSize = 50) {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+		const tasksToSync: TaskToSync[] = await this.taskRepo.getTasksToSync(Number(batchSize));
+
+		tasksToSync.forEach((task) => {
+			if (task.filerecord) {
+				this.updateFilerecordForTask(task);
+			} else {
+				const filerecord = this.createFilerecordForTask(task);
+				this.createFileFilerecord(task.file._id, filerecord._id);
+			}
+		});
+
+		await this.em.flush();
+	}
+
+	public updateFilerecordForTask(task: TaskToSync) {
+		const { filerecord } = task;
+		if (filerecord) {
+			filerecord.updatedAt = task.file.updatedAt;
+			this.em.persist(filerecord);
+		}
+	}
+
+	private createFilerecordForTask(task: TaskToSync) {
```

Comment:
I think it make sense to split the logic.

First we want to have only updated elements. Maybe it is solved by the aggregation. But we need any task.update -> for deletion and adding of files to task. For each task can exist multiple files.
-> added new file: result into -> we must create a new on for this parent
-> udate file: result into -> we must create a new on for this parent AND DELETE the already migrated
-> remove file: result into -> remove file from S3 and from meta data 

The batch of 50 task -> give us a stack from 0 to x files. Let say 120 for a example batch. 
This stack is our operation list. First we must evalute what is changed. 
-> for this we must evalute if we already have sync element
-> if yes we must add this information to our stack
-> we must add which operation is needed (delete, added, updated, only rename)

// on this place it make sense to log what operation is startet with count 

Normaly after this (but it is also possible to do it on the fly) we start with the execution of the operations.
1. delete elements + meta + S3
2. create new elements + meta + S3
3. update only name -> meta
4. update any other -> meta + S3

For all of the needed operations from legacy meta -> stack, legacy S3 -> stack, stack<-interpretation->stack, stack -> new meta, stack -> new S3 it make sense to use a class that have mapper.  The mapper can replace for each next step.

Otherwise we must rewrite in core logic for each of the up coming migration steps like submission and lessons new.